### PR TITLE
fix #1919: PROPFIND request should not redirect to GET request

### LINKED ISSUE #1919: redirect for PROFIND request issues GET request
see code in `com.squareup.okhttp.internal.http.HttpEngine#followUpRequest`

```java
      case HTTP_MULT_CHOICE:
      case HTTP_MOVED_PERM:
      case HTTP_MOVED_TEMP:
      case HTTP_SEE_OTHER:
        // Does the client allow redirects?
        if (!client.getFollowRedirects()) return null;

        String location = userResponse.header("Location");
        if (location == null) return null;
        HttpUrl url = userRequest.httpUrl().resolve(location);

        // Don't follow redirects to unsupported protocols.
        if (url == null) return null;

        // If configured, don't follow redirects between SSL and non-SSL.
        boolean sameScheme = url.scheme().equals(userRequest.httpUrl().scheme());
        if (!sameScheme && !client.getFollowSslRedirects()) return null;

        // Redirects don't include a request body.
        Request.Builder requestBuilder = userRequest.newBuilder();
        if (HttpMethod.permitsRequestBody(userRequest.method())) {
          requestBuilder.method("GET", null);
          requestBuilder.removeHeader("Transfer-Encoding");
          requestBuilder.removeHeader("Content-Length");
          requestBuilder.removeHeader("Content-Type");
        }
```

Thus, any http body is lost and out of a PROFIND request I get a GET request instead.

IMHO we need a more flexible solution for coming up with a redirected request. For PROPFIND, the redirected request should again be PROPFIND and the HTTP body should be maintained.

### PR DESCRIPTION
**Pull request by rburgst**
fix for issue #1919

### CODE REVIEW

File: okhttp/src/main/java/com/squareup/okhttp/internal/http/HttpMethod.java [hunk 1]
```diff
@@ -41,6 +41,11 @@ public static boolean permitsRequestBody(String method) {
         || method.equals("LOCK");     // (WebDAV) body: create lock, without body: refresh lock
   }
 
+  public static boolean redirectsToGET(String method) {
```
Comment (swankjesse) [inline okhttp/src/main/java/com/squareup/okhttp/internal/http/HttpMethod.java]: nit: `GET` –> `Get` ?

File: okhttp/src/main/java/com/squareup/okhttp/internal/http/HttpMethod.java [hunk 2]
```diff
@@ -41,6 +41,11 @@ public static boolean permitsRequestBody(String method) {
...
+    // all requests but propfind should redirect to a GET request
```
Comment (swankjesse) [inline okhttp/src/main/java/com/squareup/okhttp/internal/http/HttpMethod.java]: http://nedbatchelder.com/blog/201401/comments_should_be_sentences.html

Comment (rburgst) [inline okhttp/src/main/java/com/squareup/okhttp/internal/http/HttpMethod.java]: will fix this but the lines above in the same file did not really adhere to that, therefore, I didn't know that this was in the code style.

**Comment by rburgst** [comment]
implemented the review comments

**Comment by swankjesse** [comment]
Great! Thanks.

**Comment by swankjesse** [comment]
Merged manually!

**Pull request closed without merging**